csvcubed/csvcubed/cli/inspect/inspectdatasetmanager.py

```diff
@@ -0,0 +1,188 @@
+"""
+Inspect Dataset Manager
+-----------------------
+
+Collection of functions handling csv-related operations used in the inspect cli.
+"""
+
+
+import logging
+from pathlib import Path
+from typing import List, Optional
+import pandas as pd
+
+
+from csvcubed.models.inspectsparqlresults import (
+    QubeComponentResult,
+)
+from csvcubed.utils.qb.components import (
+    ComponentField,
+    ComponentPropertyAttributeURI,
+    ComponentPropertyType,
+    get_component_property_as_relative_path,
+)
+from csvcubed.models.inspectdataframeresults import (
+    DatasetObservationsByMeasureUnitInfoResult,
+    DatasetObservationsInfoResult,
+    DatasetSingleMeasureResult,
+)
+from csvcubed.models.csvcubedexception import (
+    CsvToDataFrameLoadFailedException,
+    InvalidNumberOfRecordsException,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+def _filter_components_from_dsd(
+    components: List[QubeComponentResult],
+    field: ComponentField,
+    filter: str,
+) -> List[QubeComponentResult]:
+    """
+    Filters the components for the given filter.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `List[QubeComponentResult]` - filtered results.
+    """
+
+    return [
+        component
+        for component in components
+        if component.as_dict()[field.value] == filter
+    ]
+
+
+def load_csv_to_dataframe(json_path: Path, csv_path: Path) -> pd.DataFrame:
+    """
+    Loads the csv in given path to a Panda Dataframe.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DataFrame` - Dataframe of the csv.
+    """
+
+    try:
+        dataset_path = json_path.parent / csv_path
```

Comment:
Muazzam has added some functionality to read CSVs in a common way (so we don't treat a bunch of things as `pd.nan`).

Could you make sure to use [csvcubed.utils.pandas.read_csv](https://github.com/GSS-Cogs/csvcubed/blob/main/csvcubed/csvcubed/utils/pandas.py#L15) here for consistency?